apps/ledger-live-mobile/src/families/shared/StakingErrors/NotEnoughFundFeesAlert/index.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useCallback, useMemo } from "react";
+import { useNavigation } from "@react-navigation/core";
+import { StackNavigationProp } from "@react-navigation/stack";
+import { useSelector } from "react-redux";
+import { Trans, useTranslation } from "react-i18next";
+import { Alert, Box, Flex, Text } from "@ledgerhq/native-ui";
+import { useSettings, useAccountUnit } from "~/hooks";
+import { NavigatorName, ScreenName } from "~/const";
+import { track } from "~/analytics";
+import { shallowAccountsSelector } from "~/reducers/accounts";
+import { useFetchCurrencyAll } from "@ledgerhq/live-common/exchange/swap/hooks/index";
+import { useRampCatalog } from "@ledgerhq/live-common/platform/providers/RampCatalogProvider/useRampCatalog";
+import { getAccountCurrency, getParentAccount } from "@ledgerhq/live-common/account/index";
+import { formatCurrencyUnit } from "@ledgerhq/live-common/currencies/index";
+import { AccountLike } from "@ledgerhq/types-live";
+
+type Props = {
+  account: AccountLike;
+};
+
+const NotEnoughFundFeesAlert = ({ account }: Props) => {
+  const { t } = useTranslation();
+  const { locale } = useSettings();
+  const navigation = useNavigation();
+  const accounts = useSelector(shallowAccountsSelector);
+
+  const unit = useAccountUnit(account);
+  const { data: swapAvailableIds } = useFetchCurrencyAll();
+  const { isCurrencyAvailable: isCurrencyBuyable } = useRampCatalog();
+
+  const parentAccount = getParentAccount(account, accounts);
+  const currency = getAccountCurrency(account);
+
+  const canBeSwapped = !!currency && swapAvailableIds.includes(currency.id);
+  const canBeBought = !!currency && isCurrencyBuyable(currency.id, "onRamp");
+
+  const assetName = unit.code;
+  const currentBalance = formatCurrencyUnit(unit, account.spendableBalance, {
+    showCode: true,
+    locale: locale,
+  });
+
+  const mapActionButtons = useMemo(
+    () => ({
+      [ScreenName.ExchangeBuy]: "buy",
+      [ScreenName.SwapTab]: "swap",
+      [ScreenName.ReceiveConfirmation]: "deposit",
+    }),
+    [],
+  );
+
+  const onNavigate = useCallback(
+    (screenName: keyof typeof mapActionButtons, navigatorName: string, params?: object) => {
+      track("button_clicked", {
+        button: mapActionButtons[screenName],
+        asset: currency.name,
+        source: "UndelegateFlowScreen",
+      });
+      (navigation as StackNavigationProp<{ [key: string]: object | undefined }>).navigate(
```

Comment:
Can we avoid using the `as` keyword ?
For example like this : 
<img width="674" alt="Screenshot 2024-10-24 at 18 43 35" src="https://github.com/user-attachments/assets/1bb34cd5-9d00-4738-9d85-52cf2dec0431">

Reply:
Done ✅